airflow/bin/cli.py

```diff
@@ -424,6 +424,50 @@ def clear(args):
         confirm_prompt=not args.no_confirm,
         include_subdags=not args.exclude_subdags)
 
+def mark_success(args):
+    dag = get_dag(args)
+    dagruns = dag.get_dagruns(args.start_date, args.end_date)
+    # todo: run.dag is transient, needs to be set
```

Comment:
There seem to be a few places in the code where this is manually set so I'm leaving it out of the scope of this PR.